### PR TITLE
Fix events in StructParse

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
Fix events not working in StructParse.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** #7066

### CODE REVIEW

File: src/main/java/ch/njol/skript/test/runner/StructParse.java [hunk 1]
```diff
@@ -100,7 +84,14 @@ public boolean load() {
 		try (RetainingLogHandler handler = SkriptLogger.startRetainingLog()) {
 			String structureSectionNodeKey = ScriptLoader.replaceOptions(structureSectionNodeToParse.getKey());
 			String error = "Can't understand this structure: " + structureSectionNodeKey;
-			Structure.parse(structureSectionNodeKey, structureSectionNodeToParse, error);
+			Structure structure = Structure.parse(structureSectionNodeKey, structureSectionNodeToParse, error);
+
+			if (structure != null) {
+				structure.preLoad();
+				structure.load();
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/test/runner/StructParse.java:91]: We need to make sure that preLoad did not return false. postLoad must do the same for load

File: src/main/java/ch/njol/skript/test/runner/StructParse.java [hunk 2]
```diff
@@ -100,7 +85,13 @@ public boolean load() {
 		try (RetainingLogHandler handler = SkriptLogger.startRetainingLog()) {
 			String structureSectionNodeKey = ScriptLoader.replaceOptions(structureSectionNodeToParse.getKey());
 			String error = "Can't understand this structure: " + structureSectionNodeKey;
-			Structure.parse(structureSectionNodeKey, structureSectionNodeToParse, error);
+			Structure structure = Structure.parse(structureSectionNodeKey, structureSectionNodeToParse, error);
+
+			getParser().setCurrentStructure(structure);
+			if (structure != null && (!structure.preLoad() || !structure.load() || !structure.postLoad())) {
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/StructParse.java:91]: should it be `(structure == null || (...))`?

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/test/runner/StructParse.java (hunk 2 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/test/runner/StructParse.java:91]: you may also need to remove current structure if failed, not sure

**Review by sovdeeth — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]
looks good

**Review by Pikachu920 — APPROVED** [review]
thanks!!

File: src/test/skript/tests/regressions/pull-7067-events in structs.sk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+parse:
+	results: {StructEvent::parse::*}
+	code:
+		on script load:
+			stop
+
+test "events in structs":
+	assert {StructEvent::parse::*} is not set with "using event in struct caused an error"
```
Comment (TheLimeGlass) [inline src/test/skript/tests/regressions/pull-7067-events in structs.sk:8]: Can this return a value to ensure that the parse section parsed?

**Pull request merged**